evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -9,6 +9,16 @@
 from evalml.model_family import ModelFamily
 from evalml.pipelines.utils import _make_stacked_ensemble_pipeline
 
+_ESTIMATOR_FAMILY_ORDER = [
+    ModelFamily.LINEAR_MODEL,
+    ModelFamily.DECISION_TREE,
+    ModelFamily.EXTRA_TREES,
+    ModelFamily.RANDOM_FOREST,
+    ModelFamily.XGBOOST,
+    ModelFamily.LIGHTGBM,
+    ModelFamily.CATBOOST
```

Comment:
Linear models should definitely go first. And I feel fine putting xgboost, lightgbm and catboost last.

I'm not sure we can say much a priori about the sort order between decision tree, extratrees and random forest at this time.

Current model families not shown here, which will therefore be pushed to the end of the first batch:
* K_NEIGHBORS: not currently included in automl
* SVM: not currently included in automl
* ARIMA
* ENSEMBLE: won't be included by iterative algo
* BASELINE: won't be included by iterative algo
* NONE: none of our estimators should have this label